src/Nowruz/modules/contract/components/sliderOngoing/useSliderOngoing.tsx

```diff
@@ -40,50 +43,65 @@ export const useSliderOngoing = (contract: Contract) => {
   const handleComplete = async () => {
     setOpenAlert(false);
     try {
-      dispatch(
-        updateStatus({
-          type: identityType,
-          paymentType: contract.project.payment_type,
-          id: contract.id,
-          offerStatus: 'CLOSED',
-          missionStatus: 'COMPLETE',
-        }),
-      );
       if (contract.mission) completeMission(contract.mission.id);
+      if (filter === 'ongoing') {
```

Comment:
you can create a function with if/else clase and pass missionStatus as ('COMPLETE', 'CANCELED', 'KICKED_OUT') and call this function in each handler